File: 프로그래머스/lv4/64063. 호텔 방 배정/호텔 방 배정.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;
using ll = long long;

unordered_map<ll, ll> parents;
ll Find(const ll &x) {
    if(parents[x] == 0) {
        parents[x] = x + 1;
        return x;
    }
    
    return parents[x] = Find(parents[x]);
}

vector<ll> solution(long long k, vector<ll> room_number) {
    vector<ll> answer;
    for(const ll &room : room_number)
        answer.emplace_back(Find(room));
    return answer;
}

File: 프로그래머스/4/64063. 호텔 방 배정/호텔 방 배정.py
import sys
sys.setrecursionlimit(10**6)

parents = {}

def find(x: int) -> int:
    if x not in parents:
        parents[x] = x + 1
        return x
    
    parents[x] = find(parents[x])
    return parents[x]

def solution(k, room_number):
    answer = []
    for room in room_number:
        answer.append(find(room))
    return answer